src/routeGeneration/routeGenerator.ts

```diff
@@ -143,19 +141,20 @@ export class RouteGenerator {
           properties[property.name] = this.buildPropertySchema(property);
         });
       }
-      const modelSchema = {
+
+      models[name] = {
```

Comment:
This is another stylistic change. Code should only be changed if it either fixes a known issue (that has an issue number in tsoa’s github) or if it enhances tsoa (and therefore has a tsoa issue number with the discussed enhancement).

Reply:
(Whoops I see I'm responding to your comments in reverse order, you probably want the read my comment below (the long one) before this.)

> This is another stylistic change.

I definitely would not call this a stylistic choice. I removed an additional (completely unnecessary) variable to make the code more concise and readable (at least in my opinion). You could argue that it doesn't improve these things, but I definitely wouldn't call it a stylistic choice.

> Code should only be changed if it either fixes a known issue (that has an issue number in tsoa’s github) or if it enhances tsoa (and therefore has a tsoa issue number with the discussed enhancement).

Does improving the code readability/accessibility fall under 'enhancement'? Then I definitely agree with you. 
I could make an issue called "The codebase is confusing" but I don't really see the point in doing that *if I already have some small changes laying around that improves it*. If I were to do a bigger cleanup I would have created an issue. In hindsight I *maybe* should have done that for my typeResolver cleanup, but my original idea would be to only fix those missing type guards. Unfortunately I discovered a lot of weird typing (mistakes) and I just sortof, well, fell into the rabbit hole.
I think you're massively overestimating the size of this cleanup, 95% is just typing fixes.